jetty-server/src/main/java/org/eclipse/jetty/server/Dispatcher.java

```diff
@@ -296,26 +297,23 @@ public Object getAttribute(String key)
         public Set<String> getAttributeNameSet()
         {
             HashSet<String> set = new HashSet<>();
-            for (String name : _attributes.getAttributeNameSet())
-            {
-                if (!name.startsWith(__INCLUDE_PREFIX) &&
-                    !name.startsWith(__FORWARD_PREFIX))
-                    set.add(name);
-            }
+            super.getAttributeNameSet().stream()
+                .filter(name -> !name.startsWith(__INCLUDE_PREFIX)) // TODO: why does this filter the __INCLUDE_PREFIX as well?
```

Comment:
Because if you do an include then forward, you need to remove the include attributes, else the target will serve them rather than the getServletPath and getPathInfo indicated resources.

It is all to do with the strange setup of includes not changing the path methods and only setting attributes.